### PR TITLE
feat(planner): select without from

### LINKED ISSUE #5163: Implement `SELECT` without `FROM` clause in new planner
In databend, we support the following query:
```sql
select 1;
```
A single-row, single-column result set will be retrieved.

This is somewhat like DUAL table in Oracle.

### PR DESCRIPTION
**Pull request by Veeupup**
Signed-off-by: Veeupup <code@tanweime.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

implement select with from in new planner

## Changelog

- New Feature

## Related Issues

Fixes #5163

### CODE REVIEW

**Review by leiysky — APPROVED** [review]
LGTM

File: query/src/sql/planner/binder/select.rs [hunk 1]
```diff
@@ -101,6 +103,32 @@ impl Binder {
         Ok(output_context)
     }
 
+    pub(super) async fn bind_one_table(&mut self, stmt: &SelectStmt) -> Result<BindContext> {
+        for select_target in &stmt.select_list {
+            if let SelectTarget::QualifiedName(names) = select_target {
+                for indirect in names {
+                    if indirect == &Indirection::Star {
+                        return Err(ErrorCode::LogicalError(
```
Comment (leiysky) [inline query/src/sql/planner/binder/select.rs:111]: Better to use `SemanticError` here.

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 2]
```diff
@@ -65,4 +65,10 @@ drop table t;
 drop table t1;
 drop table t2;
 
+-- Select without from
```
[the remark this answers is not in the document]
Comment (Veeupup) [reply tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:69]: yeah, will be done now

[the remark this answers is not in the document]
Comment (BohuTANG) [reply tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:69]: It seems we should split some small tests for v2 in some conflicts PR, name them like `20_0001_planner_v2_select_without_from.sql`? cc @leiysky @xudong963

[the remark this answers is not in the document]
Comment (BohuTANG) [reply tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:69]: When planner v2 is ready, we can backport and merge them to our other stateless tests.

File: query/src/sql/planner/binder/select.rs (hunk 1 above)
Comment (Veeupup) [reply query/src/sql/planner/binder/select.rs:111]: make sense

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (leiysky) [reply tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:69]: > It seems we should split some small tests for v2 in some conflicts PR, name them like `20_0001_planner_v2_select_without_from.sql`? cc @leiysky @xudong963

@BohuTANG This test suite won't exist for too long, we just exploit it to make sure there is no regression of our recent work. 

We are running stateless tests with new planner enabled to find out the features we haven't implemented yet. As soon as the new planner can pass all the test suites, this file will be deprecated.

And after migrating to new planner, we are going to migrate the test suites to `sqllogic` test framework. Which means we won't put more effort on merging this with other tests.

[the remark this answers is not in the document]
Comment (xudong963) [reply tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:78]: Yes, I think the migration will be completed soon 🐛

**Review by xudong963 — APPROVED** [review]

**Comment by BohuTANG** [comment]
CI failed:
```
test parser::test_statements_in_legacy_suites ... FAILED

failures:

---- parser::test_statements_in_legacy_suites stdout ----
thread 'parser::test_statements_in_legacy_suites' panicked at 'called `Result::unwrap()` on an `Err` value: Code: 1005, displayText = error: 
  --> SQL:4:1
  |
1 | set enable_planner_v2 = 1;
2 | 
3 | select '====SELECT_FROM_NUMBERS===='
4 | select * from numbers(10);
  | ^^^^^^ expected `.`, `(`, `IS`, `NOT`, `IN`, `BETWEEN`, or 54 more ...
```
https://github.com/datafuselabs/databend/runs/6349095598?check_suite_focus=true#step:3:328


How to find CI error, search `failed`:
image

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 3]
```diff
@@ -1,23 +1,30 @@
 set enable_planner_v2 = 1;
 
+select '====SELECT_FROM_NUMBERS===='
 select * from numbers(10);
 
 -- Comparison expressions
+select '====COMPARSION===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:7]:
```suggestion
select '====COMPARSION====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
 select * from numbers(10) where number between 1 and 9 and number > 2 and number < 8 and number is not null and number = 5 and number >= 5 and number <= 5;
 
 -- Cast expression
+select '====CAST===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:11]:
```suggestion
select '====CAST====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 5]
```diff
@@ -1,23 +1,30 @@
...
 select * from numbers(10) where cast(number as string) = '5';
 
 -- Binary operator
+select '====BINARY_OPERATOR===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:15]:
```suggestion
select '====BINARY_OPERATOR====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 6]
```diff
@@ -1,23 +1,30 @@
...
 select (number + 1 - 2) * 3 / 4 from numbers(1);
 
 -- Functions
+select '====FUNCTIONS===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:19]:
```suggestion
select '====FUNCTIONS====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 7]
```diff
@@ -1,23 +1,30 @@
...
 select sin(cos(number)) from numbers(1);
 
 -- In list
+select '====IN_LIST===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:23]:
```suggestion
select '====IN_LIST====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 8]
```diff
@@ -1,23 +1,30 @@
...
 select * from numbers(5) where number in (1, 3);
 
 -- Aggregator operator
+select '====AGGREGATER===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:27]:
```suggestion
select '====AGGREGATER====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 9]
```diff
@@ -45,6 +52,7 @@ SELECT max(number) FROM numbers_mt (10) where number > 99999999998;
 SELECT max(number) FROM numbers_mt (10) where number > 2;
 
 -- Inner join
+select '====INNER_JOIN===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:55]:
```suggestion
select '====INNER_JOIN====';
```

File: tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql [hunk 10]
```diff
@@ -67,4 +75,11 @@ drop table t2;
 
 select count(*) from numbers(1000) as t inner join numbers(1000) as t1 on t.number = t1.number;
 
+-- Select without from
+select '====SELECT_WITHOUT_FROM===='
```
Comment (leiysky) [inline tests/suites/0_stateless/20+_others/20_0001_planner_v2.sql:79]:
```suggestion
select '====SELECT_WITHOUT_FROM====';
```

**Pull request merged**